### PR TITLE
ci: trigger multi-pool conformance workflow using ariane

### PR DESCRIPTION
**Pull request by tklauser**
Convert the multi-pool conformance workflow to be triggered by Ariane using the `/ci-multi-pool` trigger phrase.

### CODE REVIEW

**Comment by tklauser** [comment]
Successful run: https://github.com/cilium/cilium/actions/runs/6095123851

File: .github/workflows/conformance-multi-pool.yaml [hunk 1]
```diff
@@ -35,11 +65,12 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 120
     steps:
-      - name: Checkout main branch to access local actions
-        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+      - name: Checkout context ref (trusted)
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
         with:
-          ref: ${{ github.event.repository.default_branch }}
+          ref: ${{ inputs.context-ref || github.sha }}
```
Comment (bimmlerd) [inline .github/workflows/conformance-multi-pool.yaml:71]: what does the `|| github.sha` mean in this case? If the `context-ref` is empty (can it be?) we fall back to checking out the SHA of the PR? Is that potentially unsafe if one somehow can force context-ref to be empty?

**Review by bimmlerd — COMMENTED** [review]
generally LGTM, one question

File: .github/workflows/conformance-multi-pool.yaml (hunk 1 above)
Comment (tklauser) [reply .github/workflows/conformance-multi-pool.yaml:71]: `context-ref` is empty if the workflow isn't triggered on a PR but by a scheduled run, ref. https://github.com/cilium/cilium/pull/27957/files#diff-083aeca09a7733ba0341688cafa9223cc3a723c9894795d5e9a8371fc606707eR20-R22 `github.sha` will always point to a SHA on `main` in this case.

On PR runs `context-ref` is always required to be set: https://github.com/cilium/cilium/pull/27957/files#diff-083aeca09a7733ba0341688cafa9223cc3a723c9894795d5e9a8371fc606707eR10-R12

Comment (bimmlerd) [reply .github/workflows/conformance-multi-pool.yaml:71]: Ah, makes sense. Thanks for the explanation!

**Review by bimmlerd — APPROVED** [review]

**Review by gandro — APPROVED** [review]

**Review by viktor-kurchenko — APPROVED** [review]

**Comment by tklauser** [comment]
Looks like I forgot to add the `commit-status-start` job which leads to the check not showing up on PRs until the workflow has finished (e.g. visible on https://github.com/cilium/cilium/pull/27968). Should be fixed by https://github.com/cilium/cilium/pull/27969.

**Pull request merged**